### ISSUE TITLE
Server error 504 when uploading metadata

### ISSUE THREAD

**Issue by changusmc** [fastlane/fastlane#16665]
Jun 24 07:31:23.362: [00:31:23]: Server error got 504

```Jun 24 07:29:04.111: Fetching gem metadata from https://rubygems.org/.........
Jun 24 07:29:04.469: Fetching gem metadata from https://rubygems.org/.
Jun 24 07:29:04.631: Resolving dependencies...
Jun 24 07:29:04.723: Fetching CFPropertyList 3.0.2
Jun 24 07:29:04.846: Installing CFPropertyList 3.0.2
Jun 24 07:29:04.878: Using public_suffix 4.0.5 (was 2.0.5)
Jun 24 07:29:04.878: Using addressable 2.7.0
Jun 24 07:29:04.879: Using atomos 0.1.3
Jun 24 07:29:04.879: Using aws-eventstream 1.1.0
Jun 24 07:29:04.881: Fetching aws-partitions 1.322.0
Jun 24 07:29:05.004: Installing aws-partitions 1.322.0
Jun 24 07:29:05.026: Fetching aws-sigv4 1.1.4
Jun 24 07:29:05.133: Installing aws-sigv4 1.1.4
Jun 24 07:29:05.151: Using jmespath 1.4.0
Jun 24 07:29:05.157: Fetching aws-sdk-core 3.97.0
Jun 24 07:29:05.435: Installing aws-sdk-core 3.97.0
Jun 24 07:29:05.845: Fetching aws-sdk-kms 1.32.0
Jun 24 07:29:05.993: Installing aws-sdk-kms 1.32.0
Jun 24 07:29:06.024: Fetching aws-sdk-s3 1.67.0
Jun 24 07:29:06.217: Installing aws-sdk-s3 1.67.0
Jun 24 07:29:06.404: Using babosa 1.0.3
Jun 24 07:29:06.404: Using bundler 2.1.4
Jun 24 07:29:06.404: Using claide 1.0.3
Jun 24 07:29:06.404: Using colored 1.2
Jun 24 07:29:06.404: Using colored2 3.1.2
Jun 24 07:29:06.404: Using highline 1.7.10
Jun 24 07:29:06.405: Using commander-fastlane 4.4.6
Jun 24 07:29:06.405: Using declarative 0.0.10
Jun 24 07:29:06.405: Using declarative-option 0.1.0
Jun 24 07:29:06.405: Using digest-crc 0.5.1
Jun 24 07:29:06.405: Using unf_ext 0.0.7.7
Jun 24 07:29:06.405: Using unf 0.1.4
Jun 24 07:29:06.405: Using domain_name 0.5.20190701
Jun 24 07:29:06.406: Using dotenv 2.7.5
Jun 24 07:29:06.406: Using multipart-post 2.0.0
Jun 24 07:29:06.407: Fetching faraday 1.0.0
Jun 24 07:29:06.555: Installing faraday 1.0.0
Jun 24 07:29:06.790: Using jwt 2.1.0
Jun 24 07:29:06.790: Using multi_json 1.14.1
Jun 24 07:29:06.790: Using multi_xml 0.6.0
Jun 24 07:29:06.791: Fetching rack 2.2.2
Jun 24 07:29:06.958: Installing rack 2.2.2
Jun 24 07:29:07.128: Fetching oauth2 1.4.4
Jun 24 07:29:07.253: Installing oauth2 1.4.4
Jun 24 07:29:07.344: Fetching dropbox_api 0.1.18
Jun 24 07:29:07.556: Installing dropbox_api 0.1.18
Jun 24 07:29:08.724: Using emoji_regex 1.0.1
Jun 24 07:29:08.724: Using excon 0.75.0 (was 0.73.0)
Jun 24 07:29:08.724: Using http-cookie 1.0.3
Jun 24 07:29:08.724: Using faraday-cookie_jar 0.0.6
Jun 24 07:29:08.725: Using faraday_middleware 1.0.0
Jun 24 07:29:08.725: Using fastimage 2.1.7
Jun 24 07:29:08.725: Using gh_inspector 1.1.3
Jun 24 07:29:08.725: Using memoist 0.16.2
Jun 24 07:29:08.726: Using os 1.1.0
Jun 24 07:29:08.726: Using signet 0.14.0
Jun 24 07:29:08.726: Using googleauth 0.13.0 (was 0.12.0)
Jun 24 07:29:08.726: Using httpclient 2.8.3
Jun 24 07:29:08.726: Using mini_mime 1.0.2
Jun 24 07:29:08.727: Using uber 0.1.0
Jun 24 07:29:08.727: Using representable 3.0.4
Jun 24 07:29:08.727: Using retriable 3.1.2
Jun 24 07:29:08.728: Using google-api-client 0.38.0
Jun 24 07:29:08.728: Using google-cloud-env 1.3.2
Jun 24 07:29:08.728: Using google-cloud-errors 1.0.1
Jun 24 07:29:08.728: Using google-cloud-core 1.5.0
Jun 24 07:29:08.728: Using google-cloud-storage 1.26.2
Jun 24 07:29:08.728: Using json 2.3.0
Jun 24 07:29:08.729: Using mini_magick 4.10.1
Jun 24 07:29:08.729: Using plist 3.5.0
Jun 24 07:29:08.729: Using rubyzip 1.3.0
Jun 24 07:29:08.730: Using security 0.1.3
Jun 24 07:29:08.730: Using naturally 2.2.0
Jun 24 07:29:08.730: Using simctl 1.6.8
Jun 24 07:29:08.730: Using slack-notifier 2.3.2
Jun 24 07:29:08.730: Using terminal-notifier 2.0.0
Jun 24 07:29:08.731: Using unicode-display_width 1.7.0
Jun 24 07:29:08.731: Using terminal-table 1.8.0
Jun 24 07:29:08.731: Using tty-screen 0.8.0
Jun 24 07:29:08.731: Using tty-cursor 0.7.1
Jun 24 07:29:08.732: Using tty-spinner 0.9.3
Jun 24 07:29:08.732: Using word_wrap 1.0.0
Jun 24 07:29:08.732: Using nanaimo 0.2.6
Jun 24 07:29:08.733: Fetching xcodeproj 1.17.0 (was 1.16.0)
Jun 24 07:29:08.889: Installing xcodeproj 1.17.0 (was 1.16.0)
Jun 24 07:29:09.052: Using rouge 2.0.7
Jun 24 07:29:09.052: Using xcpretty 0.3.0
Jun 24 07:29:09.053: Using xcpretty-travis-formatter 1.0.0
Jun 24 07:29:09.053: Using fastlane 2.150.0.rc1 (was 2.148.1)
Jun 24 07:29:09.054: Fetching rchardet 1.8.0
Jun 24 07:29:09.222: Installing rchardet 1.8.0
Jun 24 07:29:09.327: Fetching git 1.7.0
Jun 24 07:29:09.558: Installing git 1.7.0
Jun 24 07:29:09.611: Fetching xml-simple 1.1.5
Jun 24 07:29:09.716: Installing xml-simple 1.1.5
Jun 24 07:29:09.724: Fetching fastlane-plugin-bugsnag 1.4.2
Jun 24 07:29:09.845: Installing fastlane-plugin-bugsnag 1.4.2
Jun 24 07:29:09.897: Fetching fastlane-plugin-dropbox 0.3.0
Jun 24 07:29:10.005: Installing fastlane-plugin-dropbox 0.3.0
Jun 24 07:29:10.025: Using fastlane-plugin-dropbox_internal 0.3.1 from source at `../fastlane-plugin-dropbox_internal`
Jun 24 07:29:10.034: Bundle complete! 4 Gemfile dependencies, 81 gems now installed.
Jun 24 07:29:10.034: Use `bundle info [gemname]` to see where a bundled gem is installed.
Jun 24 07:29:13.454: +------------------------+---------+------------------------+
Jun 24 07:29:13.454: |                       Used plugins                        |
Jun 24 07:29:13.454: +------------------------+---------+------------------------+
Jun 24 07:29:13.454: | Plugin                 | Version | Action                 |
Jun 24 07:29:13.454: +------------------------+---------+------------------------+
Jun 24 07:29:13.454: | fastlane-plugin-dropb  | 0.3.0   | dropbox                |
Jun 24 07:29:13.454: | ox                     |         |                        |
Jun 24 07:29:13.454: | fastlane-plugin-dropb  | 0.3.1   | cibo buck              |
Jun 24 07:29:13.454: | ox_internal            |         | setup_for_dev_vm       |
Jun 24 07:29:13.454: |                        |         | xplat_root s3_actions  |
Jun 24 07:29:13.454: |                        |         | provisioning           |
Jun 24 07:29:13.454: |                        |         | crashlytics_with_exte  |
Jun 24 07:29:13.454: |                        |         | nsions                 |
Jun 24 07:29:13.454: |                        |         | clean_dsyms_before_up  |
Jun 24 07:29:13.454: |                        |         | load python_xcuitests  |
Jun 24 07:29:13.454: |                        |         | hypershard make        |
Jun 24 07:29:13.454: |                        |         | changes_json           |
Jun 24 07:29:13.454: | fastlane-plugin-bugsn  | 1.4.2   | upload_symbols_to_bug  |
Jun 24 07:29:13.454: | ag                     |         | snag                   |
Jun 24 07:29:13.454: |                        |         | send_build_to_bugsnag  |
Jun 24 07:29:13.454: +------------------------+---------+------------------------+
Jun 24 07:29:13.454: 
Jun 24 07:29:13.593: [00:29:13]: Sending anonymous analytics information
Jun 24 07:29:13.593: [00:29:13]: Learn more at https://docs.fastlane.tools/#metrics
Jun 24 07:29:13.593: [00:29:13]: No personal or sensitive data is sent.
Jun 24 07:29:13.593: [00:29:13]: You can disable this by adding `opt_out_usage` at the top of your Fastfile
Jun 24 07:29:13.615: [00:29:13]: ---------------------------
Jun 24 07:29:13.615: [00:29:13]: --- Step: opt_out_usage ---
Jun 24 07:29:13.615: [00:29:13]: ---------------------------
Jun 24 07:29:13.615: [00:29:13]: Disabled upload of used actions
Jun 24 07:29:13.615: [00:29:13]: ----------------------------------------
Jun 24 07:29:13.615: [00:29:13]: --- Step: Verifying fastlane version ---
Jun 24 07:29:13.615: [00:29:13]: ----------------------------------------
Jun 24 07:29:13.616: [00:29:13]: Your fastlane version 2.150.0.rc1 matches the minimum requirement of 2.144.0  ✅
Jun 24 07:29:13.616: [00:29:13]: ------------------------------
Jun 24 07:29:13.616: [00:29:13]: --- Step: default_platform ---
Jun 24 07:29:13.616: [00:29:13]: ------------------------------
Jun 24 07:29:13.617: [00:29:13]: Driving the lane 'ios appstore_metadata_test' 🚀
Jun 24 07:29:13.789: [00:29:13]: ---------------------
Jun 24 07:29:13.789: [00:29:13]: --- Step: deliver ---
Jun 24 07:29:13.789: [00:29:13]: ---------------------
Jun 24 07:29:13.789: [00:29:13]: Login to App Store Connect (ios-appstore-metadata-test@)
Jun 24 07:29:15.432: [00:29:15]: Login successful
Jun 24 07:29:15.881: 
Jun 24 07:29:15.885: +--------------------------------------+--------------------------------------+
Jun 24 07:29:15.885: |                         deliver 2.150.0.rc1 Summary                         |
Jun 24 07:29:15.885: +--------------------------------------+--------------------------------------+
Jun 24 07:29:15.885: | app_identifier                       | com.foo.bar            |
Jun 24 07:29:15.885: | metadata_path                        | fastlane/metadata_to_test            |
Jun 24 07:29:15.885: | screenshots_path                     | fastlane/metadata_to_test            |
Jun 24 07:29:15.885: | overwrite_screenshots                | true                                 |
Jun 24 07:29:15.885: | skip_binary_upload                   | true                                 |
Jun 24 07:29:15.885: | app_version                          | 1.0                                  |
Jun 24 07:29:15.885: | build_number                         | 1.0.0                                |
Jun 24 07:29:15.885: | ignore_language_directory_validatio  | true                                 |
Jun 24 07:29:15.885: | n                                    |                                      |
Jun 24 07:29:15.885: | force                                | true                                 |
Jun 24 07:29:15.885: | username                             | ******  |
Jun 24 07:29:15.885: |                                      | com                                  |
Jun 24 07:29:15.885: | platform                             | ios                                  |
Jun 24 07:29:15.885: | edit_live                            | false                                |
Jun 24 07:29:15.885: | use_live_version                     | false                                |
Jun 24 07:29:15.885: | skip_screenshots                     | false                                |
Jun 24 07:29:15.885: | skip_metadata                        | false                                |
Jun 24 07:29:15.885: | skip_app_version_update              | false                                |
Jun 24 07:29:15.885: | submit_for_review                    | false                                |
Jun 24 07:29:15.885: | reject_if_possible                   | false                                |
Jun 24 07:29:15.885: | automatic_release                    | false                                |
Jun 24 07:29:15.885: | phased_release                       | false                                |
Jun 24 07:29:15.885: | reset_ratings                        | false                                |
Jun 24 07:29:15.885: | run_precheck_before_submit           | true                                 |
Jun 24 07:29:15.885: | precheck_default_rule_level          | warn                                 |
Jun 24 07:29:15.885: | individual_metadata_items            | []                                   |
Jun 24 07:29:15.885: | precheck_include_in_app_purchases    | true                                 |
Jun 24 07:29:15.885: +--------------------------------------+--------------------------------------+
Jun 24 07:29:15.885: 
Jun 24 07:29:15.885: [00:29:15]: Making sure the latest version on App Store Connect matches '1.0'...
Jun 24 07:29:17.009: [00:29:16]: '1.0' is the latest version on App Store Connect
Jun 24 07:29:18.731: [00:29:18]: Loading 'fastlane/metadata_to_test/da/description.txt'...
Jun 24 07:29:18.731: [00:29:18]: Loading 'fastlane/metadata_to_test/da/keywords.txt'...
Jun 24 07:29:18.731: [00:29:18]: Loading 'fastlane/metadata_to_test/da/release_notes.txt'...
Jun 24 07:29:18.731: [00:29:18]: Loading 'fastlane/metadata_to_test/da/support_url.txt'...
Jun 24 07:29:18.732: [00:29:18]: Loading 'fastlane/metadata_to_test/da/marketing_url.txt'...
Jun 24 07:29:18.732: [00:29:18]: Loading 'fastlane/metadata_to_test/da/name.txt'...
Jun 24 07:29:18.732: [00:29:18]: Loading 'fastlane/metadata_to_test/da/subtitle.txt'...
Jun 24 07:29:18.732: [00:29:18]: Loading 'fastlane/metadata_to_test/da/privacy_url.txt'...
Jun 24 07:29:18.733: [00:29:18]: Loading 'fastlane/metadata_to_test/de-DE/description.txt'...
Jun 24 07:29:18.733: [00:29:18]: Loading 'fastlane/metadata_to_test/de-DE/keywords.txt'...
Jun 24 07:29:18.733: [00:29:18]: Loading 'fastlane/metadata_to_test/de-DE/release_notes.txt'...
Jun 24 07:29:18.733: [00:29:18]: Loading 'fastlane/metadata_to_test/de-DE/support_url.txt'...
Jun 24 07:29:18.733: [00:29:18]: Loading 'fastlane/metadata_to_test/de-DE/marketing_url.txt'...
Jun 24 07:29:18.734: [00:29:18]: Loading 'fastlane/metadata_to_test/de-DE/name.txt'...
Jun 24 07:29:18.734: [00:29:18]: Loading 'fastlane/metadata_to_test/de-DE/subtitle.txt'...
Jun 24 07:29:18.734: [00:29:18]: Loading 'fastlane/metadata_to_test/de-DE/privacy_url.txt'...
Jun 24 07:29:18.734: [00:29:18]: Loading 'fastlane/metadata_to_test/en-US/description.txt'...
Jun 24 07:29:18.735: [00:29:18]: Loading 'fastlane/metadata_to_test/en-US/keywords.txt'...
Jun 24 07:29:18.735: [00:29:18]: Loading 'fastlane/metadata_to_test/en-US/release_notes.txt'...
Jun 24 07:29:18.735: [00:29:18]: Loading 'fastlane/metadata_to_test/en-US/support_url.txt'...
Jun 24 07:29:18.735: [00:29:18]: Loading 'fastlane/metadata_to_test/en-US/marketing_url.txt'...
Jun 24 07:29:18.735: [00:29:18]: Loading 'fastlane/metadata_to_test/en-US/name.txt'...
Jun 24 07:29:18.736: [00:29:18]: Loading 'fastlane/metadata_to_test/en-US/subtitle.txt'...
Jun 24 07:29:18.736: [00:29:18]: Loading 'fastlane/metadata_to_test/en-US/privacy_url.txt'...
Jun 24 07:29:18.736: [00:29:18]: Loading 'fastlane/metadata_to_test/es-ES/description.txt'...
Jun 24 07:29:18.736: [00:29:18]: Loading 'fastlane/metadata_to_test/es-ES/keywords.txt'...
Jun 24 07:29:18.736: [00:29:18]: Loading 'fastlane/metadata_to_test/es-ES/release_notes.txt'...
Jun 24 07:29:18.737: [00:29:18]: Loading 'fastlane/metadata_to_test/es-ES/support_url.txt'...
Jun 24 07:29:18.737: [00:29:18]: Loading 'fastlane/metadata_to_test/es-ES/marketing_url.txt'...
Jun 24 07:29:18.737: [00:29:18]: Loading 'fastlane/metadata_to_test/es-ES/name.txt'...
Jun 24 07:29:18.737: [00:29:18]: Loading 'fastlane/metadata_to_test/es-ES/subtitle.txt'...
Jun 24 07:29:18.738: [00:29:18]: Loading 'fastlane/metadata_to_test/es-ES/privacy_url.txt'...
Jun 24 07:29:18.738: [00:29:18]: Loading 'fastlane/metadata_to_test/es-MX/description.txt'...
Jun 24 07:29:18.738: [00:29:18]: Loading 'fastlane/metadata_to_test/es-MX/keywords.txt'...
Jun 24 07:29:18.738: [00:29:18]: Loading 'fastlane/metadata_to_test/es-MX/release_notes.txt'...
Jun 24 07:29:18.739: [00:29:18]: Loading 'fastlane/metadata_to_test/es-MX/support_url.txt'...
Jun 24 07:29:18.739: [00:29:18]: Loading 'fastlane/metadata_to_test/es-MX/marketing_url.txt'...
Jun 24 07:29:18.739: [00:29:18]: Loading 'fastlane/metadata_to_test/es-MX/name.txt'...
Jun 24 07:29:18.739: [00:29:18]: Loading 'fastlane/metadata_to_test/es-MX/subtitle.txt'...
Jun 24 07:29:18.739: [00:29:18]: Loading 'fastlane/metadata_to_test/es-MX/privacy_url.txt'...
Jun 24 07:29:18.740: [00:29:18]: Loading 'fastlane/metadata_to_test/fr-FR/description.txt'...
Jun 24 07:29:18.741: [00:29:18]: Loading 'fastlane/metadata_to_test/fr-FR/keywords.txt'...
Jun 24 07:29:18.741: [00:29:18]: Loading 'fastlane/metadata_to_test/fr-FR/release_notes.txt'...
Jun 24 07:29:18.741: [00:29:18]: Loading 'fastlane/metadata_to_test/fr-FR/support_url.txt'...
Jun 24 07:29:18.741: [00:29:18]: Loading 'fastlane/metadata_to_test/fr-FR/marketing_url.txt'...
Jun 24 07:29:18.741: [00:29:18]: Loading 'fastlane/metadata_to_test/fr-FR/name.txt'...
Jun 24 07:29:18.741: [00:29:18]: Loading 'fastlane/metadata_to_test/fr-FR/subtitle.txt'...
Jun 24 07:29:18.741: [00:29:18]: Loading 'fastlane/metadata_to_test/fr-FR/privacy_url.txt'...
Jun 24 07:29:18.742: [00:29:18]: Loading 'fastlane/metadata_to_test/id/description.txt'...
Jun 24 07:29:18.742: [00:29:18]: Loading 'fastlane/metadata_to_test/id/keywords.txt'...
Jun 24 07:29:18.742: [00:29:18]: Loading 'fastlane/metadata_to_test/id/release_notes.txt'...
Jun 24 07:29:18.742: [00:29:18]: Loading 'fastlane/metadata_to_test/id/support_url.txt'...
Jun 24 07:29:18.742: [00:29:18]: Loading 'fastlane/metadata_to_test/id/marketing_url.txt'...
Jun 24 07:29:18.743: [00:29:18]: Loading 'fastlane/metadata_to_test/id/name.txt'...
Jun 24 07:29:18.743: [00:29:18]: Loading 'fastlane/metadata_to_test/id/subtitle.txt'...
Jun 24 07:29:18.743: [00:29:18]: Loading 'fastlane/metadata_to_test/id/privacy_url.txt'...
Jun 24 07:29:18.743: [00:29:18]: Loading 'fastlane/metadata_to_test/it/description.txt'...
Jun 24 07:29:18.744: [00:29:18]: Loading 'fastlane/metadata_to_test/it/keywords.txt'...
Jun 24 07:29:18.744: [00:29:18]: Loading 'fastlane/metadata_to_test/it/release_notes.txt'...
Jun 24 07:29:18.744: [00:29:18]: Loading 'fastlane/metadata_to_test/it/support_url.txt'...
Jun 24 07:29:18.744: [00:29:18]: Loading 'fastlane/metadata_to_test/it/marketing_url.txt'...
Jun 24 07:29:18.745: [00:29:18]: Loading 'fastlane/metadata_to_test/it/name.txt'...
Jun 24 07:29:18.745: [00:29:18]: Loading 'fastlane/metadata_to_test/it/subtitle.txt'...
Jun 24 07:29:18.745: [00:29:18]: Loading 'fastlane/metadata_to_test/it/privacy_url.txt'...
Jun 24 07:29:18.745: [00:29:18]: Loading 'fastlane/metadata_to_test/ja/description.txt'...
Jun 24 07:29:18.746: [00:29:18]: Loading 'fastlane/metadata_to_test/ja/keywords.txt'...
Jun 24 07:29:18.746: [00:29:18]: Loading 'fastlane/metadata_to_test/ja/release_notes.txt'...
Jun 24 07:29:18.746: [00:29:18]: Loading 'fastlane/metadata_to_test/ja/support_url.txt'...
Jun 24 07:29:18.746: [00:29:18]: Loading 'fastlane/metadata_to_test/ja/marketing_url.txt'...
Jun 24 07:29:18.746: [00:29:18]: Loading 'fastlane/metadata_to_test/ja/name.txt'...
Jun 24 07:29:18.747: [00:29:18]: Loading 'fastlane/metadata_to_test/ja/subtitle.txt'...
Jun 24 07:29:18.747: [00:29:18]: Loading 'fastlane/metadata_to_test/ja/privacy_url.txt'...
Jun 24 07:29:18.747: [00:29:18]: Loading 'fastlane/metadata_to_test/ko/description.txt'...
Jun 24 07:29:18.747: [00:29:18]: Loading 'fastlane/metadata_to_test/ko/keywords.txt'...
Jun 24 07:29:18.748: [00:29:18]: Loading 'fastlane/metadata_to_test/ko/release_notes.txt'...
Jun 24 07:29:18.748: [00:29:18]: Loading 'fastlane/metadata_to_test/ko/support_url.txt'...
Jun 24 07:29:18.748: [00:29:18]: Loading 'fastlane/metadata_to_test/ko/marketing_url.txt'...
Jun 24 07:29:18.748: [00:29:18]: Loading 'fastlane/metadata_to_test/ko/name.txt'...
Jun 24 07:29:18.748: [00:29:18]: Loading 'fastlane/metadata_to_test/ko/subtitle.txt'...
Jun 24 07:29:18.749: [00:29:18]: Loading 'fastlane/metadata_to_test/ko/privacy_url.txt'...
Jun 24 07:29:18.749: [00:29:18]: Loading 'fastlane/metadata_to_test/ms/description.txt'...
Jun 24 07:29:18.749: [00:29:18]: Loading 'fastlane/metadata_to_test/ms/keywords.txt'...
Jun 24 07:29:18.749: [00:29:18]: Loading 'fastlane/metadata_to_test/ms/release_notes.txt'...
Jun 24 07:29:18.749: [00:29:18]: Loading 'fastlane/metadata_to_test/ms/support_url.txt'...
Jun 24 07:29:18.750: [00:29:18]: Loading 'fastlane/metadata_to_test/ms/marketing_url.txt'...
Jun 24 07:29:18.750: [00:29:18]: Loading 'fastlane/metadata_to_test/ms/name.txt'...
Jun 24 07:29:18.750: [00:29:18]: Loading 'fastlane/metadata_to_test/ms/subtitle.txt'...
Jun 24 07:29:18.750: [00:29:18]: Loading 'fastlane/metadata_to_test/ms/privacy_url.txt'...
Jun 24 07:29:18.750: [00:29:18]: Loading 'fastlane/metadata_to_test/nl-NL/description.txt'...
Jun 24 07:29:18.751: [00:29:18]: Loading 'fastlane/metadata_to_test/nl-NL/keywords.txt'...
Jun 24 07:29:18.751: [00:29:18]: Loading 'fastlane/metadata_to_test/nl-NL/release_notes.txt'...
Jun 24 07:29:18.751: [00:29:18]: Loading 'fastlane/metadata_to_test/nl-NL/support_url.txt'...
Jun 24 07:29:18.751: [00:29:18]: Loading 'fastlane/metadata_to_test/nl-NL/marketing_url.txt'...
Jun 24 07:29:18.752: [00:29:18]: Loading 'fastlane/metadata_to_test/nl-NL/name.txt'...
Jun 24 07:29:18.752: [00:29:18]: Loading 'fastlane/metadata_to_test/nl-NL/subtitle.txt'...
Jun 24 07:29:18.752: [00:29:18]: Loading 'fastlane/metadata_to_test/nl-NL/privacy_url.txt'...
Jun 24 07:29:18.752: [00:29:18]: Loading 'fastlane/metadata_to_test/no/description.txt'...
Jun 24 07:29:18.752: [00:29:18]: Loading 'fastlane/metadata_to_test/no/keywords.txt'...
Jun 24 07:29:18.753: [00:29:18]: Loading 'fastlane/metadata_to_test/no/release_notes.txt'...
Jun 24 07:29:18.753: [00:29:18]: Loading 'fastlane/metadata_to_test/no/support_url.txt'...
Jun 24 07:29:18.753: [00:29:18]: Loading 'fastlane/metadata_to_test/no/marketing_url.txt'...
Jun 24 07:29:18.753: [00:29:18]: Loading 'fastlane/metadata_to_test/no/name.txt'...
Jun 24 07:29:18.753: [00:29:18]: Loading 'fastlane/metadata_to_test/no/subtitle.txt'...
Jun 24 07:29:18.754: [00:29:18]: Loading 'fastlane/metadata_to_test/no/privacy_url.txt'...
Jun 24 07:29:18.754: [00:29:18]: Loading 'fastlane/metadata_to_test/pl/description.txt'...
Jun 24 07:29:18.754: [00:29:18]: Loading 'fastlane/metadata_to_test/pl/keywords.txt'...
Jun 24 07:29:18.754: [00:29:18]: Loading 'fastlane/metadata_to_test/pl/release_notes.txt'...
Jun 24 07:29:18.755: [00:29:18]: Loading 'fastlane/metadata_to_test/pl/name.txt'...
Jun 24 07:29:18.755: [00:29:18]: Loading 'fastlane/metadata_to_test/pl/subtitle.txt'...
Jun 24 07:29:18.755: [00:29:18]: Loading 'fastlane/metadata_to_test/pt-BR/description.txt'...
Jun 24 07:29:18.755: [00:29:18]: Loading 'fastlane/metadata_to_test/pt-BR/keywords.txt'...
Jun 24 07:29:18.755: [00:29:18]: Loading 'fastlane/metadata_to_test/pt-BR/release_notes.txt'...
Jun 24 07:29:18.756: [00:29:18]: Loading 'fastlane/metadata_to_test/pt-BR/support_url.txt'...
Jun 24 07:29:18.756: [00:29:18]: Loading 'fastlane/metadata_to_test/pt-BR/marketing_url.txt'...
Jun 24 07:29:18.756: [00:29:18]: Loading 'fastlane/metadata_to_test/pt-BR/name.txt'...
Jun 24 07:29:18.756: [00:29:18]: Loading 'fastlane/metadata_to_test/pt-BR/subtitle.txt'...
Jun 24 07:29:18.756: [00:29:18]: Loading 'fastlane/metadata_to_test/pt-BR/privacy_url.txt'...
Jun 24 07:29:18.757: [00:29:18]: Loading 'fastlane/metadata_to_test/ru/description.txt'...
Jun 24 07:29:18.757: [00:29:18]: Loading 'fastlane/metadata_to_test/ru/keywords.txt'...
Jun 24 07:29:18.757: [00:29:18]: Loading 'fastlane/metadata_to_test/ru/release_notes.txt'...
Jun 24 07:29:18.757: [00:29:18]: Loading 'fastlane/metadata_to_test/ru/support_url.txt'...
Jun 24 07:29:18.758: [00:29:18]: Loading 'fastlane/metadata_to_test/ru/marketing_url.txt'...
Jun 24 07:29:18.758: [00:29:18]: Loading 'fastlane/metadata_to_test/ru/name.txt'...
Jun 24 07:29:18.758: [00:29:18]: Loading 'fastlane/metadata_to_test/ru/subtitle.txt'...
Jun 24 07:29:18.758: [00:29:18]: Loading 'fastlane/metadata_to_test/ru/privacy_url.txt'...
Jun 24 07:29:18.759: [00:29:18]: Loading 'fastlane/metadata_to_test/sv/description.txt'...
Jun 24 07:29:18.759: [00:29:18]: Loading 'fastlane/metadata_to_test/sv/keywords.txt'...
Jun 24 07:29:18.759: [00:29:18]: Loading 'fastlane/metadata_to_test/sv/release_notes.txt'...
Jun 24 07:29:18.759: [00:29:18]: Loading 'fastlane/metadata_to_test/sv/support_url.txt'...
Jun 24 07:29:18.759: [00:29:18]: Loading 'fastlane/metadata_to_test/sv/marketing_url.txt'...
Jun 24 07:29:18.760: [00:29:18]: Loading 'fastlane/metadata_to_test/sv/name.txt'...
Jun 24 07:29:18.760: [00:29:18]: Loading 'fastlane/metadata_to_test/sv/subtitle.txt'...
Jun 24 07:29:18.760: [00:29:18]: Loading 'fastlane/metadata_to_test/sv/privacy_url.txt'...
Jun 24 07:29:18.760: [00:29:18]: Loading 'fastlane/metadata_to_test/th/description.txt'...
Jun 24 07:29:18.761: [00:29:18]: Loading 'fastlane/metadata_to_test/th/keywords.txt'...
Jun 24 07:29:18.761: [00:29:18]: Loading 'fastlane/metadata_to_test/th/release_notes.txt'...
Jun 24 07:29:18.761: [00:29:18]: Loading 'fastlane/metadata_to_test/th/support_url.txt'...
Jun 24 07:29:18.761: [00:29:18]: Loading 'fastlane/metadata_to_test/th/marketing_url.txt'...
Jun 24 07:29:18.762: [00:29:18]: Loading 'fastlane/metadata_to_test/th/name.txt'...
Jun 24 07:29:18.762: [00:29:18]: Loading 'fastlane/metadata_to_test/th/subtitle.txt'...
Jun 24 07:29:18.762: [00:29:18]: Loading 'fastlane/metadata_to_test/th/privacy_url.txt'...
Jun 24 07:29:18.762: [00:29:18]: Loading 'fastlane/metadata_to_test/uk/description.txt'...
Jun 24 07:29:18.763: [00:29:18]: Loading 'fastlane/metadata_to_test/uk/keywords.txt'...
Jun 24 07:29:18.763: [00:29:18]: Loading 'fastlane/metadata_to_test/uk/release_notes.txt'...
Jun 24 07:29:18.763: [00:29:18]: Loading 'fastlane/metadata_to_test/uk/name.txt'...
Jun 24 07:29:18.763: [00:29:18]: Loading 'fastlane/metadata_to_test/uk/subtitle.txt'...
Jun 24 07:29:18.764: [00:29:18]: Loading 'fastlane/metadata_to_test/zh-Hans/description.txt'...
Jun 24 07:29:18.764: [00:29:18]: Loading 'fastlane/metadata_to_test/zh-Hans/keywords.txt'...
Jun 24 07:29:18.764: [00:29:18]: Loading 'fastlane/metadata_to_test/zh-Hans/release_notes.txt'...
Jun 24 07:29:18.764: [00:29:18]: Loading 'fastlane/metadata_to_test/zh-Hans/support_url.txt'...
Jun 24 07:29:18.764: [00:29:18]: Loading 'fastlane/metadata_to_test/zh-Hans/marketing_url.txt'...
Jun 24 07:29:18.765: [00:29:18]: Loading 'fastlane/metadata_to_test/zh-Hans/name.txt'...
Jun 24 07:29:18.765: [00:29:18]: Loading 'fastlane/metadata_to_test/zh-Hans/subtitle.txt'...
Jun 24 07:29:18.765: [00:29:18]: Loading 'fastlane/metadata_to_test/zh-Hans/privacy_url.txt'...
Jun 24 07:29:18.765: [00:29:18]: Loading 'fastlane/metadata_to_test/zh-Hant/description.txt'...
Jun 24 07:29:18.766: [00:29:18]: Loading 'fastlane/metadata_to_test/zh-Hant/keywords.txt'...
Jun 24 07:29:18.766: [00:29:18]: Loading 'fastlane/metadata_to_test/zh-Hant/release_notes.txt'...
Jun 24 07:29:18.766: [00:29:18]: Loading 'fastlane/metadata_to_test/zh-Hant/support_url.txt'...
Jun 24 07:29:18.766: [00:29:18]: Loading 'fastlane/metadata_to_test/zh-Hant/marketing_url.txt'...
Jun 24 07:29:18.767: [00:29:18]: Loading 'fastlane/metadata_to_test/zh-Hant/name.txt'...
Jun 24 07:29:18.767: [00:29:18]: Loading 'fastlane/metadata_to_test/zh-Hant/subtitle.txt'...
Jun 24 07:29:18.767: [00:29:18]: Loading 'fastlane/metadata_to_test/zh-Hant/privacy_url.txt'...
Jun 24 07:29:18.767: [00:29:18]: Loading 'fastlane/metadata_to_test/copyright.txt'...
Jun 24 07:29:18.768: [00:29:18]: Loading 'fastlane/metadata_to_test/primary_category.txt'...
Jun 24 07:29:18.768: [00:29:18]: Loading 'fastlane/metadata_to_test/secondary_category.txt'...
Jun 24 07:29:18.768: [00:29:18]: Loading 'fastlane/metadata_to_test/primary_first_sub_category.txt'...
Jun 24 07:29:18.768: [00:29:18]: Loading 'fastlane/metadata_to_test/primary_second_sub_category.txt'...
Jun 24 07:29:18.768: [00:29:18]: Loading 'fastlane/metadata_to_test/secondary_first_sub_category.txt'...
Jun 24 07:29:18.769: [00:29:18]: Loading 'fastlane/metadata_to_test/secondary_second_sub_category.txt'...
Jun 24 07:29:22.493: [00:29:22]: Will begin uploading metadata for '1.0' on App Store Connect
Jun 24 07:29:22.493: [00:29:22]: Skipping 'release_notes'... this is the first version of the app
Jun 24 07:29:22.494: [00:29:22]: Uploading metadata to App Store Connect for localized version 'en-US'
Jun 24 07:29:22.746: [00:29:22]: Uploading metadata to App Store Connect for localized version 'ko'
Jun 24 07:29:22.997: [00:29:22]: Uploading metadata to App Store Connect for localized version 'th'
Jun 24 07:29:23.265: [00:29:23]: Uploading metadata to App Store Connect for localized version 'uk'
Jun 24 07:31:23.362: +------------------+----------------------------+
Jun 24 07:31:23.362: |                 Lane Context                  |
Jun 24 07:31:23.362: +------------------+----------------------------+
Jun 24 07:31:23.362: | DEFAULT_PLATFORM | ios                        |
Jun 24 07:31:23.362: | PLATFORM_NAME    | ios                        |
Jun 24 07:31:23.362: | LANE_NAME        | ios appstore_metadata_test |
Jun 24 07:31:23.362: +------------------+----------------------------+
Jun 24 07:31:23.362: [00:31:23]: Server error got 504
Jun 24 07:31:23.385: 
Jun 24 07:31:23.386: +------+---------------------+-------------+
Jun 24 07:31:23.386: |             fastlane summary             |
Jun 24 07:31:23.386: +------+---------------------+-------------+
Jun 24 07:31:23.386: | Step | Action              | Time (in s) |
Jun 24 07:31:23.386: +------+---------------------+-------------+
Jun 24 07:31:23.386: | 1    | opt_out_usage       | 0           |
Jun 24 07:31:23.386: | 2    | Verifying fastlane  | 0           |
Jun 24 07:31:23.386: |      | version             |             |
Jun 24 07:31:23.386: | 3    | default_platform    | 0           |
Jun 24 07:31:23.386: | 💥   | deliver             | 129         |
Jun 24 07:31:23.386: +------+---------------------+-------------+
Jun 24 07:31:23.386: 
Jun 24 07:31:23.387: [00:31:23]: fastlane finished with errors
Jun 24 07:31:23.441: 
Jun 24 07:31:23.441: Looking for related GitHub issues on fastlane/fastlane...
Jun 24 07:31:23.441: 
Jun 24 07:31:24.076: ➡️  Version getters
Jun 24 07:31:24.076:     https://github.com/fastlane/fastlane/pull/504 [closed] 4 💬
Jun 24 07:31:24.077:     04 Feb 2017
Jun 24 07:31:24.077: 
Jun 24 07:31:24.077: ➡️  fastlane deliver: Potential server error received: 'Unexpected Error'
Jun 24 07:31:24.077:     https://github.com/fastlane/fastlane/issues/16621 [open] 116 💬
Jun 24 07:31:24.077:     a day ago
Jun 24 07:31:24.077: 
Jun 24 07:31:24.077: ➡️  [deliver][produce][spaceship] Update to use new App Store Connect API endpoints
Jun 24 07:31:24.077:     https://github.com/fastlane/fastlane/pull/16626 [closed] 136 💬
Jun 24 07:31:24.077:     a day ago
Jun 24 07:31:24.077: 
Jun 24 07:31:24.077: and 1 more at: https://github.com/fastlane/fastlane/search?q=The%20request%20could%20not%20be%20completed%20because%3A%0A%09Server%20error%20got%20504&type=Issues&utf8=✓
Jun 24 07:31:24.077: 
Jun 24 07:31:24.077: 🔗  You can ⌘ + double-click on links to open them directly in your browser.```

**Comment by joshdholtz** [comment]
@changusmc Hey! I've been sporadically seeing this on my side as well. It seems to be something with the API because it also sometimes happens on the App Store Connect website. I've gotten around it by running fastlane again 😬 I have code locally to retry a request when a 504 returns but I haven't been able to get a 504 lately 😔 

Do you get the 504 every time?

**Comment by changusmc** [comment]
I've seen it sporadically locally, didn't have a good repro.

It seems to be more frequent in CI.

I can add retries as well, thanks for the suggestion. Do you plan on adding retries within deliver?

**Comment by joshdholtz** [comment]
@changusmc I just got a 504 on my side! I just added a bunch of more languages to my test app and that seemed to trigger it. Working on adding retries right now 😊

**Comment by clindsay** [comment]
@changusmc @joshdholtz I am getting it reliably when I try to upload screenshots. It never happens at the same time, but I have 168 screenshots to upload, and I've never gotten passed about 20-30 of them before hitting a 504. If there's a way to workaround this and have it retry when encountering a 504, I'd love to know about it!

**Comment by joshdholtz** [comment]
@clindsay @changusmc Can either of you do this for me? 👇 I'd like to see what _is_ 504ing for you.

Run `bundle exec fastlane run spaceship_logs print_contents:true` and uploaded the output to me? It will output the contents from our Spaceship/API logs so you can see what the last requests/responses were like. It can sometimes help with seeing more of the error response 🙂

**Comment by clindsay** [comment]
@joshdholtz Is the `bundle exec` required? 

```
Ultros:StudyJapanese chris$ bundle exec fastlane run spaceship_logs print_contents:true
-bash: /usr/local/bin/bundle: /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby: bad interpreter: No such file or directory
```

I ran it without that, and got this: https://www.dropbox.com/s/2f8eg7hknopyy8t/spaceship1593008155_95978_70246576890060.log?dl=0

**Comment by joshdholtz** [comment]
@clindsay Thank you! I think I got a fix that should help you then 😊 Are you using the release candidate right now?

**Comment by clindsay** [comment]
@joshdholtz Awesome! Yes, I am.

**Comment by joshdholtz** [comment]
Cool cool! I will have this fix in the next release candidate for you 😊 The fix I got should look allow up to 5 retries on a POST and PATCH when a 504 is received.

I can put this in a new branch right now if you are able to switch from the release candidate to a specific branch. Are you using a `Gemfile`?

**Comment by clindsay** [comment]
@joshdholtz I'm not, but I'm fine switching to one.

**Comment by clindsay** [comment]
@joshdholtz OK, I am now using a Gemfile 👍

**Comment by joshdholtz** [comment]
Perfect! I will get that branch made here in about 5 minutes. Just getting of a phone call 😊

**Comment by changusmc** [comment]
I am using the release candidate. Do you still need my logs or are you good to go?

I'm reminded now that we are uploading lots of screenshots as well. Our metadata hasn't changed since Apple changed their endpoint.

**Comment by changusmc** [comment]
We are using a Gemfile as well

**Comment by joshdholtz** [comment]
@changusmc @clindsay Created #16667 that _should_ help 🤞 

1. Update `Gemfile` to below 👇 
2. Run `bundle install`, `bundle update`, or `bundle update fastlane`

```rb
gem "fastlane", :git => "https://github.com/fastlane/fastlane.git", :branch => "joshdholtz-2.150.0-thread-delete-screenshot-and-retry-504"
```

**Comment by changusmc** [comment]
> ```ruby
> ```ruby
> gem "fastlane", :git => "https://github.com/fastlane/fastlane.git", :branch => "joshdholtz-2.150.0-thread-delete-screenshot-and-retry-504"
> ```
> 
> 
>       
> ```
Thanks for the quick fix!

I kicked off a CI job with these changes

**Comment by changusmc** [comment]
I think it worked! It failed further down in the pipeline, should I create a separate issue?

```
Jun 24 16:59:25.036: [09:59:25]: An attribute value is not acceptable for the current resource state. - The attribute 'releaseType' can not be modified.
```

https://gist.github.com/changusmc/fb2502dbb4b492c1d584946c776be2dd

**Comment by joshdholtz** [comment]
@changusmc We can just add that one here 😊 Can you do this for me? 👇 

Run bundle exec fastlane run spaceship_logs print_contents:true and uploaded the output to me? It will output the contents from our Spaceship/API logs so you can see what the last requests/responses were like. It can sometimes help with seeing more of the error response 🙂

**Comment by clindsay** [comment]
@changusmc @joshdholtz mine is still running, but seems to be working so far.

**Comment by clindsay** [comment]
@joshdholtz Finished, all good! Almost every screenshot took about 3 seconds to upload, and then every 30-40 one would take like 90 seconds. But it finished successfully! Thank you so much!

**Comment by joshdholtz** [comment]
@clindsay @changusmc Did either of you see something like `"Timeout received! Retrying after 3 seconds (remaining: #{tries})..."` in your logs? That will show when the retry happen 😇 Just curious!

**Comment by clindsay** [comment]
@joshdholtz I didn't. Is it really possible I just lucked out with this run? Bizarre if so, since I've tried 4-5 times previously with no luck.

**Comment by changusmc** [comment]
> Timeout received!

I also didn't see it, I'll trigger a few runs.

**Comment by mohsinahd** [comment]
I got following error after updating to 
`Failed to delete screenshot - en-US APP_IPAD_PRO_129 03da4ff7-9758-48ac-ab5e-00145603b679`

after updating to 
`gem "fastlane", :git => "https://github.com/fastlane/fastlane.git", :branch => "joshdholtz-2.150.0-thread-delete-screenshot-and-retry-504"
`

**Comment by joshdholtz** [comment]
Totally possible 😛

**Comment by mohsinahd** [comment]
I deleted the image manually and then it started to work .. ..

**Comment by joshdholtz** [comment]
@mohsinahd I will add a retry around `DELETE` as. well! That is probably what happened

**Comment by joshdholtz** [comment]
@mohsinahd I lied... Delete was already there 😱 Hmmmm... Can you run this? 👇 

Run bundle exec fastlane run spaceship_logs print_contents:true and uploaded the output to me? It will output the contents from our Spaceship/API logs so you can see what the last requests/responses were like. It can sometimes help with seeing more of the error response 🙂

**Comment by joshdholtz** [comment]
I got it to show for me! This prevent my screenshot uploads from failing halfway through

```sh
INFO [2020-06-24 13:16:31.64]: Uploading './fastlane/screenshots/en-US/iPad Pro (12.9-inch) (3rd generation)-0Launch.png'...
Timeout received! Retrying after 3 seconds (remaining: 4)...
INFO [2020-06-24 13:18:59.42]: Uploading './fastlane/screenshots/en-US/iPad Pro (12.9-inch) (3rd generation)-1Launch.png'...
```

**Comment by mohsinahd** [comment]
Is there any plan to run upload snapshots in parallel threads to make the process efficient ?

**Comment by joshdholtz** [comment]
@mohsinahd Ha! I actually just had this conversation with someone 😊 Adding concurrency can always make things harder but I did add it to delete so I _think_ we maybe should add it to upload. But...

We don't want too many operations going at once. We don't have any concept of "thread pool" in _fastlane_ right now so we could pseudo thread pool by multi threading inside of a single screenshot set (meaning iPad Pro 12' 9", iPhone 5' 5", etc). It should help a little bit and then we can add proper thread pooling later on down the road.

**Comment by joshdholtz** [comment]
Nevermind. I added a thread pool. I'm making this happen.

**Comment by mohsinahd** [comment]
Sounds great, when should we expect stable version with all these amazing improvements all to-gather?

**Comment by joshdholtz** [comment]
I'm planning having a new release candidate (2.150.0.rc2) out later today and having new stable version out end of week (pending feedback 😇 )

**Comment by joshdholtz** [comment]
I lied. This pooling is getting weird 😔 Will have to punt for a bit.

**Comment by mohsinahd** [comment]
It's alright for now, DELETE snapshot pooling thing has improved the overall meta uploading process efficient a bit.

**Comment by changusmc** [comment]
I got the full spaceship logs but will include the pieces that I think are relevant

```

Jun 24 18:40:38.447: DEBUG [11:38:48]: << PATCH appInfoLocalizations/bfce49b9-83cb-4299-8377-b499f0a6c2af: 504 <HTML><HEAD>
--
Jun 24 18:40:38.447: <TITLE>Gateway Timeout - In read </TITLE>
Jun 24 18:40:38.447: </HEAD><BODY>
Jun 24 18:40:38.447: <H1>Gateway Timeout</H1>
Jun 24 18:40:38.447: The proxy server did not receive a timely response from the upstream server.<P>
Jun 24 18:40:38.447: Reference&#32;&#35;1&#46;3def6368&#46;1593023928&#46;ae224e8
Jun 24 18:40:38.447: </BODY></HTML>
Jun 24 18:40:38.447:
Jun 24 18:40:38.447: WARN  [11:38:48]: Timeout received! Retrying after 3 seconds (remaining: 4)...


Jun 24 18:40:38.447: DEBUG [11:40:33]: << PATCH appStoreVersions/da86bbe2-9df0-4246-8008-d4f71ef80e52: 409 {"errors"=>[{"id"=>"b86cc99a-7b7b-49e7-8a8b-223c7039bdc0", "status"=>"409", "code"=>"ENTITY_ERROR.ATTRIBUTE.INVALID.INVALID_STATE", "title"=>"An attribute value is not acceptable for the current resource state.", "detail"=>"The attribute 'releaseType' can not be modified.", "source"=>{"pointer"=>"/data/attributes/releaseType"}}]}
--



```

**Comment by joshdholtz** [comment]
@changusmc What kind of state is your app in right now? Is it in "Prepare for Submission"? Its not liking the setting of "Release Type" which is either "Schedule", "After Approval", or "Manual".?

**Comment by changusmc** [comment]
> @changusmc What kind of state is your app in right now? Is it in "Prepare for Submission"? Its not liking the setting of "Release Type" which is either "Schedule", "After Approval", or "Manual" and I'm not sure why 🤔

It's set to "Manual"

**Comment by joshdholtz** [comment]
@changusmc Is your app in "Prepare for Submission" right now?

**Comment by changusmc** [comment]
Yes it's in "Prepare for Submission"

**Comment by joshdholtz** [comment]
Weird! I'm not sure why you would get that error then 😔 Can you try again?

**Comment by changusmc** [comment]
> Weird! I'm not sure why you would get that error then 😔 Can you try again?

Got this error twice so far. 

I do see this new option in ITC, i'll select it and try again.

![image](https://user-images.githubusercontent.com/8715059/85621986-0572c000-b61b-11ea-89b5-8dd9e320a9d1.png)

**Comment by changusmc** [comment]
Got the same error again.

**Comment by changusmc** [comment]
![image](https://user-images.githubusercontent.com/8715059/85625635-9f893700-b620-11ea-9f80-d5736819b0a2.png)
1. Everything looks to be in order for me

2. We never want this app released. We use this to sample app test metadata for our production apps.

**Comment by koxon** [comment]
I also see 500 and 504 sporadically.... Apple reliability in display.
Something that bother me though is that I can't submit for review, I get the error mentioned in the issue: 
https://github.com/fastlane/fastlane/issues/16663

Every submission I do I get:

[12:54:18]: FastlaneCore::BuildWatcher found more than 1 matching build:
1.0.30(767) for IOS - VALID
1.0.12(767) for IOS - VALID
[...]

**Comment by joshdholtz** [comment]
@koxon The 504 should be fixed in `2.150.0.rc2` 🤞 We have extended the timeout and also added in some retries.

But... can you create a new issue of your 500? I would like to see the output and error to see if there is some logic we can add in to prevent that. Thanks! ❤️

**Comment by joshdholtz** [comment]
Closing this issue since this is fixed in https://github.com/fastlane/fastlane/pull/16640

**Comment by changusmc** [comment]
@joshdholtz should I create a new issue for https://github.com/fastlane/fastlane/issues/16665#issuecomment-649019837

**Comment by joshdholtz** [comment]
@changusmc Yes please! If you could include that whole spaceship log that would be great 😊 If you don’t want to post publicly here you can email it to me at me@joshholtz.com 💪

**Comment by changusmc** [comment]
Was a new issue created for `500`?

I'm currently on rc4 and only uploading text metadata. I can send more logs in email. 

I don't see any retries in my spaceship logs. I'm running `bundle exec fastlane run spaceship_logs print_contents:true` in CI whenever there's an non-zero exit code.

```

```Jun 29 19:33:46.117: DEBUG [12:31:58]: << PATCH appStoreVersionLocalizations/af30726c-33ed-4c93-a811-571db2e0d489: 504 <HTML><HEAD>
Jun 29 19:33:46.117: <TITLE>Gateway Timeout - In read </TITLE>
Jun 29 19:33:46.117: </HEAD><BODY>
Jun 29 19:33:46.117: <H1>Gateway Timeout</H1>
Jun 29 19:33:46.117: The proxy server did not receive a timely response from the upstream server.<P>
Jun 29 19:33:46.117: Reference&#32;&#35;1&#46;20e8d93f&#46;1593459118&#46;1bffcf38
Jun 29 19:33:46.117: </BODY></HTML>
Jun 29 19:33:46.117: 
Jun 29 19:33:46.117: DEBUG [12:33:42]: << PATCH appStoreVersionLocalizations/af30726c-33ed-4c93-a811-571db2e0d489: 500 {"errors"=>[{"status"=>"500", "code"=>"UNEXPECTED_ERROR", "title"=>"An unexpected error occurred.", "detail"=>"An unexpected error occurred on the server side. If this issue continues, contact us at https://developer.apple.com/contact/."}]}``````

**Comment by joshdholtz** [comment]
@changusmc Are you still receiving this 500?

**Comment by changusmc** [comment]
> @changusmc Are you still receiving this 500?

Yes I got it on our post commit job, so it looks like a flake. We are now on rc4.

I didn't see any retries in the logs.

**Comment by joshdholtz** [comment]
@changusmc Ah weird, so this 500 doesn't happen every time for you?

**Comment by changusmc** [comment]
> @changusmc Ah weird, so this 500 doesn't happen every time for you?

We don't have enough sample size yet, we don't run this CI build on every commit, only when metadata file paths change and once a week.

The job with rc4 has passed 4/5 times. The weird part is the failing build didn't have any retry logic. I sent you the full logs in email.

**Comment by joshdholtz** [comment]
@changusmc I only have retry on 504s right now but I can add some retries on 500 😊 I'll have that included in rc5

**Comment by changusmc** [comment]
Ah ok, well that explains it :)

**Comment by joshdholtz** [comment]
I've only experienced some flakieness on 504s but might as well add it to 500s 🤷‍♂️

**Comment by joshdholtz** [comment]
@changusmc PR https://github.com/fastlane/fastlane/pull/16713 should have fixes for you! I will release this as rc5 in about 30 minutes

**Comment by joshdholtz** [comment]
`rc5` is out if you'd like to see if that solves what you were seeing!

**Comment by joshdholtz** [comment]
Closing this since this should be fixed now! Latest version is `2.150.1`
